FreeRTOS-Plus/Source/FreeRTOS-IoT-Libraries-LTS-Beta2/c_sdk/platform/freertos/transport/src/plaintext_freertos.c

```diff
@@ -31,129 +31,70 @@
 #include "FreeRTOS_IP.h"
 #include "FreeRTOS_Sockets.h"
 
+/* FreeRTOS Socket wrapper include. */
+#include "freertos_sockets_wrapper.h"
+
 /* Transport interface include. */
 #include "plaintext_freertos.h"
 
-/* Maximum number of times to call FreeRTOS_recv when initiating a graceful shutdown. */
-#ifndef TRANSPORT_FREERTOS_SHUTDOWN_LOOPS
-    #define TRANSPORT_FREERTOS_SHUTDOWN_LOOPS    ( 3 )
-#endif
-
-/* A negative error code indicating a network failure. */
-#define TRANSPORT_FREERTOS_NETWORK_ERROR    ( -1 )
-
-BaseType_t Plaintext_FreeRTOS_Connect( NetworkContext_t * pNetworkContext,
-                                       const char * pHostName,
-                                       uint16_t port,
-                                       uint32_t receiveTimeoutMs,
-                                       uint32_t sendTimeoutMs )
+PlaintextTransportStatus_t Plaintext_FreeRTOS_Connect( NetworkContext_t * pNetworkContext,
+                                                       const char * pHostName,
+                                                       uint16_t port,
+                                                       uint32_t receiveTimeoutMs,
+                                                       uint32_t sendTimeoutMs )
 {
-    Socket_t tcpSocket = FREERTOS_INVALID_SOCKET;
+    PlaintextTransportStatus_t plaintextStatus = PLAINTEXT_TRANSPORT_SUCCESS;
     BaseType_t socketStatus = 0;
-    struct freertos_sockaddr serverAddress = { 0 };
-    TickType_t transportTimeout = 0;
-
-    /* Create a new TCP socket. */
-    tcpSocket = FreeRTOS_socket( FREERTOS_AF_INET, FREERTOS_SOCK_STREAM, FREERTOS_IPPROTO_TCP );
 
-    if( tcpSocket == FREERTOS_INVALID_SOCKET )
+    if( ( pNetworkContext == NULL ) || ( pHostName == NULL ) )
     {
-        LogError( ( "Failed to create new socket." ) );
-        socketStatus = TRANSPORT_FREERTOS_NETWORK_ERROR;
+        LogError( ( "Invalid input parameter(s): Arguments cannot be NULL. pNetworkContext=%p, "
```

Comment:
GCC gives warnings for printing pointers of struct type to with `%p` format specifier that represents the `void *` type. Thus, had to create a PR https://github.com/aws/aws-iot-device-sdk-embedded-C/pull/1109 for adding casts to `void *` on C-SDK. Does Visual Studio not show any warnings?

Reply:
Ok. Visual studio doesn't show any warnings for this